### PR TITLE
feat: update built-in proxy

### PR DESCRIPTION
**Pull request by Henvy-Mango**
# Tittle Here
update built-in proxy

## Description
feat: allow set custom provider for built-in proxy
feat: update tcp2wsServer

add guide in websocket setting

## Check List

- [x] I have tested the changes and verified that they work and don't break anything(as well as I can manage) or drop the support for previous versions.
- [x] My code follows the code style of this project
- [x] I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

### CODE REVIEW

File: TMessagesProj/src/main/res/values/strings_nullgram.xml [hunk 1]
```diff
@@ -218,9 +218,12 @@
     <string name="autoDisableBuiltInProxyDesc">It will automatically turn off the built-in proxy when Nullgram detected you turn on the VPN</string>
     <string name="NekogramWsDescription">Connect to Telegram through WebSocket via Cloudflare.\nProvided by Nekogram</string>
     <string name="NekogramXWsDescription">Provided by Nekogram X</string>
+    <string name="WsGetHelp">Get Help</string>
+    <string name="WsCustomDescription">Use custom domain to connect to Telegram through WebSocket via Cloudflare.</string>
```
Comment (qwq233) [inline TMessagesProj/src/main/res/values/strings_nullgram.xml:222]: 要输出正常的telegram要用TELEOFFICIAL

**Review by qwq233 — COMMENTED** [review]
能不能加个说明 教别人怎么自建代理 用TYPE_INFO_PRIVACY就行

**Comment by Henvy-Mango** [comment]
> 能不能加个说明 教别人怎么自建代理 用TYPE_INFO_PRIVACY就行

TYPE_INFO_PRIVACY不能复制链接，自建代理需要导入dns记录的文件

现在是跳转到Nekogram它的帮助频道
photo_2023-04-20_15-35-03

**Comment by qwq233** [comment]
> > 能不能加个说明 教别人怎么自建代理 用TYPE_INFO_PRIVACY就行
> 
> TYPE_INFO_PRIVACY不能复制链接，自建代理需要导入dns记录的文件
> 
> 现在是跳转到Nekogram它的帮助频道 photo_2023-04-20_15-35-03

到时候我来做吧

**Pull request closed without merging**